kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/CSVRecordParser.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2013 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.data.spi.filesystem;
+
+import au.com.bytecode.opencsv.CSVParser;
+import java.io.IOException;
+import java.util.List;
+import org.kitesdk.data.DatasetIOException;
+import org.kitesdk.data.View;
+import org.kitesdk.data.spi.DataModelUtil;
+
+public class CSVRecordParser<E> {
+  private final CSVParser parser;
+  private final CSVRecordBuilder<E> builder;
+
+  public CSVRecordParser(CSVProperties props,
+                         View<E> view,
+                         List<String> header) {
+    this.parser = CSVUtil.newParser(props);
+    this.builder = new CSVRecordBuilder<E>(
+        DataModelUtil.getReaderSchema(
+            view.getType(), view.getDataset().getDescriptor().getSchema()),
+        view.getType(), header);
+  }
+
+  public E read(String line) {
```

Comment:
This should probably just call `read(String line, E reuse)` and pass in `null` for `reuse` rather than duplicate the method.